File: ClientServerExample/src/main/java/org/akka/essentials/clientserver/sample/ServerActorSystem.java
package org.akka.essentials.clientserver.sample;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;

public class ServerActorSystem {
    private LoggingAdapter log = null;
    private ActorSystem actorSystem;

    public ServerActorSystem() {
        Config config = ConfigFactory.load();
        Config serverSysConfig = config.getConfig("ServerSys");
        actorSystem = ActorSystem.create("ServerSys", serverSysConfig);
        log = Logging.getLogger(actorSystem, this);
        ActorRef actor = actorSystem.actorOf(Props.create(ServerActor.class), "serverActor");
    }

    public static void main(String[] args) {
        new ServerActorSystem();
    }
}

File: ClientServerExample/src/main/scala/org/akka/essentials/clientserver/example/ServerActorSystem.scala
package org.akka.essentials.clientserver.example

import akka.actor.{ActorSystem, Props}
import com.typesafe.config.ConfigFactory

object ServerActorSystem {
  def main(args: Array[String]): Unit = {
    val config = ConfigFactory.load().getConfig("ServerSys")
    val actorSystem = ActorSystem("ServerSys",config)
    val serverActor = actorSystem.actorOf(Props[ServerActor],name = "serverActor")
  }

}
